### DISCUSSION TITLE
Can we have Parallelism for Single Partition Export

### DISCUSSION THREAD

**Discussion by jkthedba** [mydumper/mydumper#1530] [Ideas]
I have mysql DB where single partition size is over 200GB can we have parallelism in mydumper for single partition

**Comment by davidducos** [comment]
Hi @jkthedba, 
If you have a integer primary key, then it should work. If not, file a bug with the table structure.

**Reply by jkthedba** [reply]
Does it mean if PK is integer data type then mydumper supports parallelism for single partition export but in my case im lucky PK is varchar but varchar have number only.. So how to do in my case
<img width="1130" alt="image" src="https://github.com/mydumper/mydumper/assets/59903526/b21fd3e2-57e7-4c20-8a7b-a3a586debab4">

**Reply by jkthedba** [reply]
App teams have designed all tables with PK type as varchar, table structures can i send 1:1 for data compliance guidelines

**Reply by davidducos** [reply]
We did an effort to implement varchar splitting but it was not a good approach. We might re implemented with the lessons learn but not soon. You should use bigint if you have only numbers.

**Reply by jkthedba** [reply]
thanks @davidducos , noted..let me discuss with dev teams
